main.cpp

```diff
@@ -64,64 +64,95 @@ send me a DM to check your pull request
         return nullptr;
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float value1 { 0 }, value2 { 0 };
+    float multiply(float* updatedValue)      //12
     {
-        
+        if(updatedValue != nullptr)
+        {
+          std::cout << "U's value1 value is " << this->value1 << std::endl;
+          value1 = *updatedValue;
+          std::cout << "U's value1 new value is " << this->value1 << std::endl;
+        }
+
+        while(std::abs(value1 - value2) > 0.001f)
+        {
+            if(std::abs(value2) > std::abs(value1))
+            {
+                ++value1;
+            }
+            else
+            {
+                ++value2;
+            }
+        }
+
+        std::cout << "U's value2 updated value: " << value2 << std::endl;
+
+        return value2 * value1;      
     }
+
 };
 
-struct <#structname2#>
+struct StaticStruct
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float staticFunction(U* that, float* updatedValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        std::cout << "U's value1 value: " << that->value1 << std::endl;
+        that->value1 = *updatedValue;
+        std::cout << "U's value1 updated value: " << that->value1 << std::endl;
+        while(std::abs(that->value2 - that->value1) > 0.001f)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            that->value2 +=0.1f;
         }
-        std::cout << "U's <#name2#> updated value: " << that-><#name2#> << std::endl;
-        return that-><#name2#> * that-><#name1#>;
+        std::cout << "U's value2 updated value: " << that->value2 << std::endl;
+        return that->value2 * that->value1;
     }
 };
         
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1(1.2f , "t1");                                             //6
+    T t2(-0.2f , "t2");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
+    MyStruct f;                                            //7
+    auto* smaller = f.compare(&t1 ,&t2);                       if(smaller != nullptr)      //8
     std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    else
+    std::cout << t1.name << "and" << t2.name << " are equel." << std::endl;
     
```

Comment:
- [ ] fix the indentation
- [ ] `compare` returns nullptr if they are equal AND for one other reason.  your `else` statement is not 100% accurate as a result.